### ISSUE TITLE
[C# 9.0] Naming rules gets confused with records

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#48782]
**Version Used**: 
Version 16.9.0 Preview 2.0 [30620.140.main]

Naming rules complain if its uppercase and complain if its lowercase. Its very confused.

**Steps to Reproduce**:

Use https://github.com/dotnet/project-system/blob/master/.editorconfig#L74

```
private record SpecialWebFolder(SpecialWebFolderFlag Flag, KnownMonikers Icon, KnownMonikers expandedIcon);
```
**Expected Behavior**:
???

**Actual Behavior**:
Severity	Code	Description	Project	File	Line	Suppression State
Warning	IDE1006	Naming rule violation: These words must begin with upper case characters: expandedIcon	Microsoft.VisualStudio.ProjectSystem.Managed.VS	C:\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS\ProjectSystem\VS\Web\Tree\SpecialWebFolderPropertiesProvider.cs	79	Active
Warning	IDE1006	Naming rule violation: The first word, 'Icon', must begin with a lower case character	Microsoft.VisualStudio.ProjectSystem.Managed.VS	C:\project-system\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS\ProjectSystem\VS\Web\Tree\SpecialWebFolderPropertiesProvider.cs	79	Active

**Comment by sharwell** [comment]
This claims to be fixed with #47509

**Comment by davidwengier** [comment]
Yes, this is fixed. These are treated as properties, so the first IDE1006 is expected, the second is not.

I'm not seeing it on Version 16.9.0 Preview 2.0 [30619.73.main]
![image](https://user-images.githubusercontent.com/754264/96642807-66649780-1372-11eb-945f-2839f2e2aeb0.png)

**Comment by davkean** [comment]
Still repro'ing this in Version 16.9.0 Preview 2.0 [30620.140.main]:

![Uploading image.png…]()

**Comment by davkean** [comment]
Checkout this branch: https://github.com/dotnet/project-system/tree/feature/AspNet and delete: https://github.com/dotnet/project-system/blob/feature/AspNet/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/GlobalSuppressions.cs#L76.

**Comment by davidwengier** [comment]
Wow, weird. I copied the .editorconfig file from that branch when I took my above screenshot.

**Comment by sharwell** [comment]
@davkean The project is hard coded to use a version of IDE1006 that doesn't support that case

https://github.com/dotnet/project-system/blob/b17cab2d0ffab5372e3fac6464d942576840c31e/build/import/HostAgnostic.props#L44-L45

**Comment by sharwell** [comment]
Duplicate of #47508

**Comment by davkean** [comment]
So I I upgrade to the latest build of those, this should be fixed?

**Comment by sharwell** [comment]
It should be fixed in the 16.8 preview 4 release of the analyzers.
